packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/queue/lambda/lib/queue.py

```diff
@@ -0,0 +1,71 @@
+from math import ceil
+from time import time
+import json
+
+class QueueHandler:
+  def __init__(self, ecs_client, sqs_client, environ):
+    self.ecs = ecs_client
+    self.sqs = sqs_client
+    self.cluster_name = environ['CLUSTER_NAME']
+    self.service_name = environ['SERVICE_NAME']
+    self.namespace = environ['NAMESPACE']
+    self.queue_names = str(environ['QUEUE_NAMES']).split(',')
+
+  def handler(self, event, context):
+    try:
+      runningCount = self.getRunningTaskCount()
+      backlogs = [self.getQueueBacklog(queueName, runningCount) for queueName in self.queue_names]
+      self.timestamp = int(time() * 1000)
+      for backlog in backlogs:
+        self.emitBacklogPerTaskMetric(backlog['queueName'], backlog['backlogPerTask'])
+    except Exception as e:
+      Exception('Exception: {}'.format(e))
+
+  """
+  This method writes the backlogPerTask metric to the stdout according to the Cloudwatch embedded metric format.
+  """
+  def emitBacklogPerTaskMetric(self, queueName, backlogPerTask):
+    print(json.dumps({
+      "_aws": {
+        "Timestamp": self.timestamp,
```

Comment:
Out of curiosity did we validate if the metrics appear in CW with the correct timestamp?

Reply:
Yes, they do!